### PR TITLE
Image input UI enhancements

### LINKED ISSUE #1334: UX: Improve file upload in settings
I have also quickly prototyped a UI for the image/file upload that would provide an enhanced UX. This was surprisingly easy to implement so far, but it would need some clean-up/refinement. We could do that in a separate PR or I could append it to this PR.

image

image

_Originally posted by @tillprochaska in https://github.com/tactilenews/100eyes/issues/1323#issuecomment-1114924332_

### PR DESCRIPTION
**Pull request by tillprochaska**
~Merge #1323 first!~

This PR implements to follow-ups to #1323:

* Improve the file UI in the settings to allow editors to view currently uploaded files and select files to upload via the system file picker or via drag and drop.
* Use ActiveStorage image transformations to optimize/scale down the logo and hero image displayed on the onboarding page in order to reduce loading time, especially on mobile.

https://user-images.githubusercontent.com/1512805/169650708-d9f9ac66-b63c-4e56-ac35-15e9952aa2d6.mp4

Closes #1334

### CODE REVIEW

File: app/components/image_input/image_input.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    if (!file.type.startsWith('image/')) {
+      return;
+    }
+
+    this.inputTarget.files = event.dataTransfer.files;
+    this.updateFilePreview(event.dataTransfer.files[0]);
+    this.element.focus();
+  }
+
+  updateFilePreview(file) {
```
Comment (tillprochaska) [inline app/components/image_input/image_input.js:48]: Call `revokeObjectURL` for the currently selected URL before creating a new one!

File: app/components/base_field/base_field.html.erb [hunk 2]
```diff
@@ -1,6 +1,10 @@
 <div <%= attrs %>>
   <div class="BaseField-textWrapper <%= hide_label ? 'visually-hidden' : '' %>">
-    <label class="BaseField-labelText" for="<%= id %>">
+    <label
+      class="BaseField-labelText"
+      for="<%= id %>"
+      id="label-<%= id %>"
```
Comment (roschaefer) [inline app/components/base_field/base_field.html.erb:6]: Why is this necessary?

**Review by roschaefer — APPROVED** [review]
image


image



After `yarn run build` I tried this out on my machine:

image


See my only PR comment. I can't see any reason to give an `id` to the `base_field`.

File: app/components/base_field/base_field.html.erb (hunk 2 above)
Comment (drjakob) [reply app/components/base_field/base_field.html.erb:6]: Clippy forever

Comment (tillprochaska) [reply app/components/base_field/base_field.html.erb:6]: > Why is this necessary?

It allows us to reference the label in the `aria-describedby` attribute. This ensures that screenreaders announce something like "Group: Header-Bild" when you focus the image input.
